Metro crashes often on Linux

# Problem

I am running `make run-metro` on Linux, but it crashes often:

```
/home/attila/Projects/LogosLab/status-mobile/node_modules/nullthrows/nullthrows.js:9
  throw error;
  ^

Error: Got unexpected null
    at nullthrows (/home/attila/Projects/LogosLab/status-mobile/node_modules/nullthrows/nullthrows.js:7:15)
    at Timeout.emitChange [as _onTimeout] (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/index.js:823:48)
    at listOnTimeout (node:internal/timers:569:17)
    at process.processTimers (node:internal/timers:512:7) {
  framesToPop: 1
}

Node.js v18.16.0
```

Another case:

```
node:internal/fs/watchers:255
    throw error;
    ^

Error: ENOENT: no such file or directory, watch '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm'
    at FSWatcher.<computed> (node:internal/fs/watchers:247:19)
    at Object.watch (node:fs:2343:34)
    at NodeWatcher._watchdir (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/NodeWatcher.js:145:24)
    at /home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/NodeWatcher.js:264:22
    at Walker.<anonymous> (/home/attila/Projects/LogosLab/status-mobile/node_modules/metro-file-map/src/watchers/common.js:127:31)
    at Walker.emit (node:events:513:28)
    at /home/attila/Projects/LogosLab/status-mobile/node_modules/walker/lib/walker.js:69:16
    at FSReqCallback.oncomplete (node:fs:198:23) {
  errno: -2,
  syscall: 'watch',
  code: 'ENOENT',
  path: '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm',
  filename: '/home/attila/Projects/LogosLab/status-mobile/node_modules/react-native-gesture-handler/android/build/kotlin/compileDebugKotlin/cacheable/caches-jvm/jvm'
}

Node.js v18.16.0
make: *** [Makefile:268: run-metro] Error 1

```

--- a/nix/scripts/node_modules.sh
+++ b/nix/scripts/node_modules.sh
@@ -46,7 +46,7 @@ copyNodeModules() {
   # WARNING: We can't de-reference .bin symlinks
   cp -Rf ${src}/node_modules/.bin/. "${tmp}/.bin/"
   # Cmake called by react-native-reanimated 3.3.0 during build creates read-only temporary files.
-  chmod -R u+w node_modules/react-native-reanimated/android/.cxx -f || true
+  chmod -R u+w node_modules/react-native-reanimated/android/ -f || true
   rm -r "${dst}"
   mv -f "${tmp}" "${dst}"
 }

